### ISSUE TITLE
restore error 

### ISSUE THREAD

**Issue by andressepter** [muesli/docker-backup#8]
Tried to backup and restore alfresco instance. Backup runs well, but restore ends w. error:
Restoring Container with hostname: 25347149be76
Pulling Image: alfresco-content-services-cyber-alfresco:stable
Error response from daemon: pull access denied for alfresco-content-services-cyber-alfresco, repository does not exist or may require 'docker login'
Interetingly enough it restores alfresco database (postgres) and alfresco search services, all other containers get same error when restoring.

**Comment by walkention** [comment]
Solved this in my case by running `docker pull <image-name>` before running the restore. Still an issue though.

**Comment by andressepter** [comment]
Well, I already tried it. Will possibly work w. "default" images but not custom built. 
Also tried to generate local repository w. custom built alfresco images (work in progress currently).

**Comment by walkention** [comment]
Ahh, got it. Misunderstood your setup.

**Comment by muesli** [comment]
You're right we can improve this. Currently docker-backup only stores the image's name/tag and restores that information when restoring the backup. The image may have changed (or got deleted) in the meantime though, so we should additionally offering storing the image itself, too.

**Comment by andressepter** [comment]
I have an IDEA (not tested yet)
During backup I will create my very own local repository. During restore I will add "localhost:5000" (or what ever location I coose) to the json file. Have to try it out. 
And yes - ability to backup WHOLE docker container (whatever its origin may be, a standard or self-built) and restore it with data volumes will absolutely ROCK! 
There is no such script / program out there. I have been struggiling w. full container backup / restore 2 weeks now  (Ive also re-created (ugly and clumsy) replica of your code in BASH) w. no good results.

**Comment by andressepter** [comment]
"You're right we can improve this. Currently docker-backup only stores the image's name/tag and restores that information when restoring the backup. The image may have changed (or got deleted) in the meantime though"
One addition : I already tried this scenario 1. build alfresco codebase from scratch 2. restore images OVER working codebase. -- still it tries to download images from docker.io although image lies right there locally.

**Comment by GoldraK** [comment]
I made a small scripts in bash to backup the docker-compose and Dockerfile to build the image locally when I want to restore but the same thing happens to @andressepter

Some idea??

Thanks for the time.

**Comment by andreasotto** [comment]
+1

**Comment by andreasotto** [comment]
```
Restoring Container: traefik_reverse-proxy_1
Pulling Image: traefik:v2.4
Error response from daemon: No such image: traefik:v2.4
```
?

**Comment by mauamolat** [comment]
I am getting 502 error when restoring from tar
